packages/utils/src/lib/control.ts

```diff
@@ -64,3 +64,47 @@ export function promiseWithResolve<T>(): Promise<T> & {
 		reject: reject!,
 	})
 }
+
+/** @internal */
+export class RC<out T> {
```

Comment:
can we call this `ReferenceCountWithFixedTimeout`? RC is too ambiguous :P

and also i don't want somebody to accidentally use this (and accidentally forget to set `timeout`) and it just immediately disposes things b/c the default is 0 😆